harness/determined/pytorch/_pytorch_context.py

```diff
@@ -360,6 +360,8 @@ def configure_apex_amp(
         return models, optimizers
 
     def _should_communicate_and_update(self) -> bool:
+        if not self.env.training:
```

Comment:
Why is this function called in non-training mode?

Reply:
It's called in `step_optimizer()`, which is called as part of any `train_batch`, regardless of whether or not the code is running in a task container.